.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,27 @@
+name: Run Benchmarks
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
```

Comment:
Do we need to be deliberate about hardware and OS for consistent results?

Reply:
Yea, great question. Reproducible env's was one of my follow-up tasks: https://github.com/patrickhuie19/benchmark-action?tab=readme-ov-file#open-questions

We can pin the OS with this top level yaml field:
```
runs-on: ubuntu-22.04
```

This gives us:
4 CPU + 16gb RAM

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories